src/resource.ts

```diff
@@ -0,0 +1,340 @@
+/* eslint-disable new-cap */
+import qs from 'qs';
+import { AxiosInstance, AxiosResponse } from 'axios';
+import { cloneDeep } from 'lodash';
+
+import Model from './model';
+import List from './models/List';
+
+/**
+ * @deprecated since 2.2.0
+ */
+export type ResourceCallback = (error: any, resource: any) => void;
+
+export interface IParentParams {
+  resource: string;
+  id: string;
+}
+
+/**
+ * The base resource
+ */
+export default class Resource {
+  /**
+   * Resource code such as `payments`
+   * @var {string} resource
+   */
+  public static resource: string;
+  /**
+   * Refers to a Model class
+   * @var {Model} model
+   */
+  public static model: any;
+  /**
+   * @var {string} apiName
+   */
+  public readonly apiName: string;
+  /**
+   * @var {string} resourcePrefix
+   */
+  public readonly resourcePrefix: string;
+  /**
+   * @var {AxiosInstance} httpClient
+   */
+  protected readonly httpClient: AxiosInstance;
+  /**
+   * @var {string} parentId
+   */
+  protected parentId: string;
+
+  /**
+   * Constructor
+   */
+  public constructor(httpClient: AxiosInstance) {
+    this.httpClient = httpClient;
+  }
+
+  /**
+   * Error handler
+   */
+  protected static errorHandler(response: any, cb?: Function): any {
+    const error = (response && response.data) || response;
+
+    if (cb) {
+      return cb(error);
+    }
+    return error;
+  }
+
+  /**
+   * Set the parent ID by providing the parent
+   *
+   * @since 1.1.1
+   *
+   * @deprecated This method is not available
```

Comment:
```suggestion
   * @deprecated since 2.2.0
```

Reply:
Turns out to be since v2 of the API, actually. Does that correspond to v2.0.0 of the client?

Reply:
Seems like it. Shall I change it to `@deprecated 2.2.0. This method is not supported by the v2 API.` to make it clearer?

Reply:
Please do!